src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -243,7 +246,10 @@ private CSharpSyntaxNode ParseExpressionOrPattern()
                         // X.Y.Z { ... } : PropertyPattern
                         else if (tk == SyntaxKind.OpenBraceToken)
                         {
-                            node = _syntaxFactory.PropertyPattern(type, this.ParseSubPropertyPatternList());
+                            var openBrace = this.EatToken(SyntaxKind.OpenBraceToken);
+                            var contents = this.ParseSubPropertyPatternList(ref openBrace);
+                            var closeBrace = this.EatToken(SyntaxKind.CloseBraceToken);
+                            node = _syntaxFactory.PropertyPattern(type, openBrace, contents, closeBrace);
```

Comment:
These 4 lines look exactly like the ones on lines 56-59, would it make sense to extract them into a method?